libraries/aws_ec2_image.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+require 'aws_backend'
+
+class AwsEc2Image < AwsResourceBase
+  name 'aws_ec2_image'
+  desc 'Verifies settings for an AWS EC2 image'
+
+  example "
+    describe aws_ec2_image(image_id: 'ami-12345678') do
+      it { should exist }
+    end
+  "
+
+  def initialize(opts = {})
+    opts = { image_id: opts } if opts.is_a?(String)
+    super(opts)
+    validate_parameters(required: [:image_id])
+    @display_name = opts[:image_id]
+
+    catch_aws_errors do
+      resp = @aws.compute_client.describe_images({ image_ids: [opts[:image_id]] })
+      if resp.images.first.nil?
+        empty_response_warn
+      else
+        @ami = resp.images[0].to_h
```

Comment:
does this need to be a global variable? I don't see `@ami` being used outside of this else statement

Reply:
might be nice to keep this consistent with the if statement and use `.first` _(this is a nitpick more than anything)_

Reply:
Updated in latest commit, file now called aws_ami.rb